addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/username_idor_scanner.js

```diff
@@ -0,0 +1,112 @@
+// This script passively scans the incoming WebSocket messages for usernames.
+// Finds usernames which are hashed with one of the following encoding methods {MD2, MD5
+// SHA1, SHA256, SHA384, SHA512}. The usernames has to be defined in any context.
+
+// Based on: org.zaproxy.zap.extension.pscanrulesBeta.UsernameIdorScanner
+
+// Author: Manos Kirtas (manolis.kirt@gmail.com)
+
+var Model = Java.type("org.parosproxy.paros.model.Model");
+var Control = Java.type("org.parosproxy.paros.control.Control");
+var ExtensionUserManagement = Java.type("org.zaproxy.zap.extension.users.ExtensionUserManagement");
+var DigestUtils = Java.type("org.apache.commons.codec.digest.DigestUtils");
+
+OPCODE_TEXT = 0x1;
+RISK_INFO 	= 0;
+CONFIDENCE_HIGH = 3;
+
+usersList = null;
+extUserManagment = null;
+
+function scan(helper,msg) {
+
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing || (usersList = getUsers() == null)){
+        return;
+    }
+
+    var counter = 0;
+    var message = String(msg.getReadablePayload()).valueOf();
+
+    usersList.forEach(function(user){
+
+        var username = user.getName();
+        var usernameHashes = getHashes(username);
+
+        Object.keys(usernameHashes).forEach(function(hashType){
+            var hash = new RegExp(usernameHashes[hashType]);
+            if((matches = message.match(usernameHashes[hashType]))!= null) {
+                matches.forEach(function(evidence){
+                    helper.newAlert()
+                        .setRiskConfidence(RISK_INFO, CONFIDENCE_HIGH)
+                        .setName("Username Hash Found")
+                        .setDescription(getDiscription(user, hashType))
+                        .setParam(counter++)
```

Comment:
I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

Reply:
Fixed in ee4f50813